Making max integrated cache staleness configurable for Spark connector (#32592)
Azure/azure-sdk-for-java: sdk/cosmos/azure-cosmos-spark_3_2-12/docs/configuration-reference.md

| `spark.cosmos.read.maxItemCount`                               | `1000`  | Overrides the maximum number of documents that can be returned for a single query- or change feed request. The default value is `1000` - consider increasing this only for average document sizes significantly smaller than 1KB or when projection reduces the number of properties selected in queries significantly (like when only selecting "id" of documents etc.).                                                                                                                                                                                                                                                                                         |

> | `spark.cosmos.read.maxIntegratedCacheStaleness.inMilliseconds` | None    | Sets the max time window in milliseconds for which query results remain cached in the integrated cache in the dedicated gateway connectivity mode. For more information on `MaxIntegratedCacheStaleness`, please visit this [link](https://learn.microsoft.com/azure/cosmos-db/integrated-cache)                                                                                                                                                                                                                                                                                                                                                                  |

Reviewer: ```suggestion
| `spark.cosmos.read.maxIntegratedCacheStaleness.inMilliseconds` | None    | Sets the max time window in milliseconds for which query results remain cached in the integrated cache in the dedicated gateway connectivity mode. Learn more about `MaxIntegratedCacheStaleness` [here](https://learn.microsoft.com/azure/cosmos-db/integrated-cache)                                                                                                                                                                                                                                                                                                                                                                  |
```

> | `spark.cosmos.read.maxIntegratedCacheStaleness.inHours`        | None    | Sets the max time window in hours for which query results remain cached in the integrated cache in the dedicated gateway connectivity mode. For more information on `MaxIntegratedCacheStaleness`, please visit this [link](https://learn.microsoft.com/azure/cosmos-db/integrated-cache)                                                                                                                                                                                                                                                                                                                                                                         |

Reviewer: Same documentation update for this as well and for other docs. 
Also, what happens if user sets both of them?
